code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -69,17 +69,29 @@ Contains most of the procs that are called when a mob is attacked by something
 
 	return siemens_coefficient
 
+
+/mob/living/carbon/human/proc/add_limb_armor(obj/item/armor_item)
+	for(var/i in limbs)
+		var/datum/limb/limb_to_check = i
+		if(!(limb_to_check.body_part & armor_item.flags_armor_protection))
+			continue
+		limb_to_check.armor = limb_to_check.armor.attachArmor(armor_item.armor)
+
+
+/mob/living/carbon/human/proc/remove_limb_armor(obj/item/armor_item)
+	for(var/i in limbs)
+		var/datum/limb/limb_to_check = i
+		if(!(limb_to_check.body_part & armor_item.flags_armor_protection))
+			continue
+		limb_to_check.armor = limb_to_check.armor.detachArmor(armor_item.armor)
+
+
 //this proc returns the armour value for a particular external organ.
-/mob/living/carbon/human/proc/getarmor_organ(datum/limb/def_zone, type)
-	if(!type)	return 0
-	var/protection = 0
-	var/list/protective_gear = list(head, wear_mask, wear_suit, w_uniform, gloves, shoes)
-	for(var/gear in protective_gear)
-		if(gear && istype(gear ,/obj/item/clothing))
-			var/obj/item/clothing/C = gear
-			if(C.flags_armor_protection & def_zone?.body_part)
-				protection += C.armor.getRating(type)
-	return protection
+/mob/living/carbon/human/proc/getarmor_organ(datum/limb/affected_limb, type)
+	if(!type)
+		CRASH("[src] called getarmor_organ() for [affected_limb] with no type.")
```

Comment:
this will already runtime when called without a type

Reply:
Want the check removed?

Reply:
its just extra code that doesnt actually provide extra debugging info since there would still be a stack trace.